src/lib/helpers.ts

```diff
@@ -68,44 +68,82 @@ export const mergeSimilarItems =
     }, [])
   }
 
-export const handleScrollToBibliographyItem = (view: EditorView) => {
+export const handleScrollToSelectedTarget = (view: EditorView) => {
   const tr = view.state.tr
 
   const node = tr.doc.nodeAt(tr.selection.$from.pos)
```

Comment:
Let's check if the selection is a `NodeSelection`. There should be a `isNodeSelection` function defined somewhere. We can probably get the `node` directly from the selection as well.

Reply:
I'm  sure how this could be useful, or if I'm understanding it correctly. I tried using it, but it didn't work as expected. The isNodeSelection always returned false. I also attempted to get the node using selection.$from.node(), but it didn't work well for bibliography items— it returned the bibliography element instead of the individual item, which would require additional code to retrieve the selected item. Additionally, I couldn't access node.selection directly. Instead, I implemented an update that provides a direct way to retrieve the DOM node at the current selection position without needing to handle the selection further.

Reply:
Got it.